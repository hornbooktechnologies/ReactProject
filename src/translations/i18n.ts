import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
