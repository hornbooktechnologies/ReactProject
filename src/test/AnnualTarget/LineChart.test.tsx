import React from "react";
import renderer from "react-test-renderer";
import LineGraph from "../../components/annualTarget/LineChart";
import { I18nextProvider } from "react-i18next";
import i18n from "../../i18nForTests";
describe("Component: LineGraph", () => {
  const props = {
    name: "OPEX",
    time: 1646905332,
    value: "AnnualData",
    data: {
      PlannedCurrentYear: 58.54,
      ForecastCurrentYear: 28.78,
      Annual: {
        1: { Actual: [], Estimates: [], Forecast: [], Period: [], Planned: [] },
      },
      Cumulative: {
        1: { Actual: [], Estimates: [], Forecast: [], Period: [], Planned: [] },
      },
      Daily: {
        1: { Actual: [], Estimates: [], Forecast: [], Period: [], Planned: [] },
      },
      Monthly: {
        1: { Actual: [], Estimates: [], Forecast: [], Period: [], Planned: [] },
      },
      Weekly: {
        1: { Actual: [], Estimates: [], Forecast: [], Period: [], Planned: [] },
      },
    },
  };
  it("LineGraph renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <LineGraph />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
