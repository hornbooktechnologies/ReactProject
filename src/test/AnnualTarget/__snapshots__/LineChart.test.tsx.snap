// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LineGraph LineGraph renders correctly 1`] = `null`;
